### PR TITLE
Adds custom allocator support to DeserializeFromNpyString().

### PR DESCRIPTION
**Pull request by jiawen**
Also renames absl::string_view to std::string_view since .bazelrc
specifies C++17 as the default.

### CODE REVIEW

**Comment by jiawen** [comment]
@fbleibel-g PTAL.

File: npy_array/npy_array/npy_array.cc [hunk 1]
```diff
@@ -18,14 +18,14 @@
...
+#include <string_view>
 #include <vector>
 
-#include "glog/logging.h"
 #include "absl/strings/match.h"
 #include "absl/strings/numbers.h"
 #include "absl/strings/str_join.h"
 #include "absl/strings/str_split.h"
-#include "absl/strings/string_view.h"
+#include "glog/logging.h"
```
Comment (fbleibel-g) [inline npy_array/npy_array/npy_array.cc:28]: Unless it really bothers you, let's keep its current position... The export from third_party will reset the header order.

Comment (jiawen) [reply npy_array/npy_array/npy_array.cc]: Moved back.

The rest is all my clang-format, which I think is the same as Google's internal one.

**Comment by fbleibel-g** [comment]
Just kidding, I can't just accept your PR, I need pull in your changes myself and re-export them. The headers will reset automatically. Hold on...

**Comment by fbleibel-g** [comment]
Aaand it's merged! https://github.com/google-research/google-research/commit/fe4f373668b678b6177c433013d34a88d40b4116

**Comment by jiawen** [comment]
Thanks!

**Pull request closed without merging**